File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/LoginKeyFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;

class LoginKeyFlattenedDto
{

    const PROPERTIES = [
            'loginProvider' => 'string',
            'providerKey' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var LoginKey
     */
    private $value;

    /**
     * @param LoginKey $value
     */
    public function __construct(LoginKey $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new LoginKey();
        }
    }

    /**
     * @return LoginKey
     */
    public function toLoginKey()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getLoginProvider()
    {
        return $this->value->getLoginProvider();
    }

    /**
     * @param string $loginProvider
     */
    public function setLoginProvider($loginProvider)
    {
        $this->value->setLoginProvider($loginProvider);
    }

    /**
     * @return string
     */
    public function getProviderKey()
    {
        return $this->value->getProviderKey();
    }

    /**
     * @param string $providerKey
     */
    public function setProviderKey($providerKey)
    {
        $this->value->setProviderKey($providerKey);
    }


}

File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/LoginKeyFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class LoginKeyFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "loginProvider",
            "providerKey",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private LoginKey value;

    public LoginKeyFlattenedDto()
    {
        this(new LoginKey());
    }

    public LoginKeyFlattenedDto(LoginKey value)
    {
        this.value = value;
    }

    public LoginKey toLoginKey()
    {
        return this.value;
    }


    public String getLoginProvider()
    {
        return this.value.getLoginProvider();
    }

    public void setLoginProvider(String loginProvider)
    {
        this.value.setLoginProvider(loginProvider);
    }

    public String getProviderKey()
    {
        return this.value.getProviderKey();
    }

    public void setProviderKey(String providerKey)
    {
        this.value.setProviderKey(providerKey);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        LoginKeyFlattenedDto other = (LoginKeyFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

